### ISSUE TITLE
[Feature] Expose JSHandle _remoteObject

### ISSUE THREAD

**Issue by nojvek** [microsoft/playwright#2275]
When I debug the the console messages passed via

```ts
        page.on(`console`, (msg) => {
          const msgType = msg.type();
          const args = msg.args()
          // proxy browser console stream
          console[msgType](args);
```

All I see is a bunch of JSHandles, I tried to figure out if I can get a usable value out of it, like strings, numbers, or proper Error objects with intact stack traces, but I couldn't 

I just want to proxy the console error messages from browser to node process 

how do I do this ?

EDIT:

await the jsonValue() helps but I still don't get errors 

```
      await new Promise((resolve, reject) => {
        // lint rule is very aggresive,
        // eslint-disable-next-line @typescript-eslint/no-misused-promises
        page.on(`console`, async (msg) => {
          const msgType = msg.type();
          const args = await Promise.all(msg.args().map((jsHandle) => jsHandle.jsonValue()));
          // proxy browser console stream
          // eslint-disable-next-line no-console
          console[msgType](...args);
        });
      });
```

Do you have plans to expose `_remoteObject` ? Seems like we could write a simple function that could return JS like objects for errors types.

**Comment by pavelfeldman** [comment]
The simplest way to proxy is:

```js
page.on('console', console.log);
```

If you'd like to process arguments, you can use [`arg.jsonValue()`](https://playwright.dev/#version=v1.0.2&path=docs%2Fapi.md&q=jshandlejsonvalue).

**Comment by nojvek** [comment]
I did try `arg.jsonValue()` but it gave `{}` for errors.

If anyone else hits this same problem, this is something that sort of works.

```
        let msgType = msg.type();
        msgType = msgType === `warning` ? `warn` : msgType; // so we can feed it to console.warn
        const msgText = msg.text();
        const args = msg.args().map((jsHandle) => {
          // see https://github.com/microsoft/playwright/issues/2275
          const remoteObject = jsHandle[`_remoteObject`];
          if (Object.prototype.hasOwnProperty.call(remoteObject, `value`)) {
            return remoteObject.value;
          } else if (remoteObject.type === `object` && remoteObject.subtype === `error` && remoteObject.description) {
            const errStack = remoteObject.description;
            const errMessage = errStack.split(`\n`)[0];
            const err = new Error(errMessage);
            err.stack = errStack;
            return err;
          }
          // we don't know how to parse this out yet, return as is
          return remoteObject;
        });

        // proxy browser console stream
        console[msgType](...args);
```

It turns out, getting error messages out of console in a uniform way from firefox, chromium, webkit is quite tricky.

I'll bash my head more against this to dig even deeper but ideal experience I'm looking for is rich console proxying. 

@pavelfeldman , just want to say, I'm a big fan of you from chrome-devtools -> puppeteer -> playwright days. Thanks for building amazing things. Really loving playwright. I hope I'm not keeping you awake at 1am.

If you're curious, I'm on an  experiment to see if I can make a mocha runner that `just` works across, nodejs, chromium, firefox, safari. The ability to have tests run locally on your machine and inspect them with a debugger, or have every test suite sharded and run on the cloud (other people's machines) in parallel. Iteratively change code and get immediate feedback running 1000s tests within <1m.

**Comment by pavelfeldman** [comment]
Hang on, trying to understand your scenario. We definitely don't want it to be that complicated to process errors. Is it something like below?

```js
console.log(new Error("error message"));
```

If that is the case, we definitely want to make `jsonValue` work for you and return an object with error message and stack in case of errors. Before we do that, you can do the following:

```js
const {message, stack} = await arg.evaluate(error => ({ message: error.message, stack: error.stack}));
```

What happens above is we take a handle to an error message and evaluate on it, cherry picking two property values that we return as values, not as a handle.

**Comment by pavelfeldman** [comment]
Closing as fixed in #2329
